core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -433,26 +430,14 @@ func (eb *EthBroadcaster) handleAnyInProgressEthTx(ctx context.Context, fromAddr
 	return nil, false
 }
 
-// getInProgressEthTx returns either 0 or 1 transaction that was left in
-// an unfinished state because something went screwy the last time. Most likely
-// the node crashed in the middle of the ProcessUnstartedEthTxs loop.
-// It may or may not have been broadcast to an eth node.
-func (eb *EthBroadcaster) getInProgressEthTx(fromAddress gethCommon.Address) (etx *EthTx, err error) {
-	etx = new(EthTx)
-	err = eb.q.Get(etx, `SELECT * FROM eth_txes WHERE from_address = $1 and state = 'in_progress'`, fromAddress.Bytes())
-	if errors.Is(err, sql.ErrNoRows) {
-		return nil, nil
-	} else if err != nil {
-		return nil, errors.Wrap(err, "getInProgressEthTx failed while loading eth tx")
-	}
-	if err = eb.orm.LoadEthTxAttempts(etx); err != nil {
-		return nil, errors.Wrap(err, "getInProgressEthTx failed while loading EthTxAttempts")
+// this function is used to pass the queryer from the txmgr to the keystore
+// it is inevitable we have to pass the queryer because we need none increment to happen in atomic transaction
```

Comment:
```suggestion
// it is inevitable we have to pass the queryer because we need the keystate's next nonce to be incremented atomically alongside the transition from `in_progress` to `broadcast` so it is ready for the next transaction
```